bndtools.core/src/org/bndtools/core/templating/repobased/ReposTemplateLoader.java

```diff
@@ -95,7 +96,7 @@ void activate() {
             if (workspace == null)
                 workspace = Central.getWorkspace();
             workspaceRepos = workspace.getPlugins(Repository.class);
-            tmpLocator = new RepoPluginsBundleLocator(workspace.getRepositories());
+            tmpLocator = new RepoPluginsBundleLocator(workspace.getPlugins(RepositoryPlugin.class), workspace.getPlugin(HttpClient.class));
```

Comment:
I think it would be better to delay creation of the `RepoPluginsBundleLocator` until the preference configured repo is created and include that as well. 

But with the current `RepoPluginsBundleLocator` implementation that doesn't add any value as it seems not to work at all for `OSGiRepository`